model/src/main/scala/org/broadinstitute/dsde/rawls/StringUtils.scala

```diff
@@ -0,0 +1,25 @@
+package org.broadinstitute.dsde.rawls
+
+import org.broadinstitute.dsde.rawls.model.{Attributable, AttributeName, ErrorReport, ErrorReportSource}
+import spray.http.StatusCodes
+
+trait StringUtils {
+  implicit val errorReportSource: ErrorReportSource
+
+  //in general, we only support alphanumeric, spaces, _, and - for user-input
+  def validateUserDefinedString(s: String): Unit = {
+    if(!s.matches("[A-z0-9_-]+")) throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(message = s"Invalid input: $s. Input may only contain alphanumeric characters, underscores, and dashes.", statusCode = StatusCodes.BadRequest))
```

Comment:
This will compile the regex for each call to this method -- it would be more efficient to compile it ahead of time:
```
val AlphaNum = "[A-z0-9_-]+".r
def validateUserDefinedString(s: String): Unit = {
  if (AlphaNum.pattern.matcher(s).matches) {
    // etc
  }
}
```